packages/cli/lib/deamon.js

```diff
@@ -0,0 +1,195 @@
+const urllib = require('urllib');
+const AutoLaunch = require('auto-launch');
+const path = require('node:path');
+const fs = require('node:fs/promises');
+const { chmodSync } = require('node:fs');
+const { rsBindingPath } = require('@cnpmjs/binding');
+const execa = require('execa');
+
+const {
+  baseRapidModeDir,
+  nydusd,
+  nydusdConfigFile,
+  nydusdMnt,
+  socketPath,
+  nydusdLogFile,
+} = require('./constants');
+
+const deamonDir = path.join(baseRapidModeDir(), 'project');
+
+const metadataDir = path.join(deamonDir, 'metadata');
+
+const rapidDeamon = rsBindingPath
+  ? path.join(rsBindingPath, 'rapid_deamon')
+  : undefined;
+
+
+const destinationFilePath = path.join(deamonDir, 'rapid_deamon');
+
+const daemonPoint = 'http://localhost:33889';
+const aliveUrl = `${daemonPoint}/alive`;
+const killUrl = `${daemonPoint}/kill`;
+const delUrl = `${daemonPoint}/del-project`;
+const addUrl = `${daemonPoint}/add-project`;
+
+const checkDeamonAlive = async () => {
+  try {
+    const result = await urllib.request(`${aliveUrl}`, {
+      method: 'GET',
+    });
+    return result.status === 200;
+  } catch (_) {
+    return false;
+  }
+};
+
+const delProject = async projectName => {
+  let config;
+  try {
+    await fs.stat(metadataDir);
+    const configPath = path.join(metadataDir, `${projectName}.json`);
+    const configBuffer = await fs.readFile(configPath);
```

Comment:
It's not recommended to directly read files using 'fs' and 'json parse'. You should consider the possibility of file corruption. It's better to wrap these operations with additional safeguards.